doc/md/guides/mysql/check-constraint.md

```diff
@@ -0,0 +1,321 @@
+---
+id: check-constraint
+title: CHECK Constraint in MySQL
+slug: /guides/mysql/check-constraint
+---
+
+### Introduction
+
+A `CHECK` constraint is a very useful tool that can be used in MySQL to prevent invalid data from being inserted into a table.
+
+It works by checking the values of certain columns in the table against certain conditions that you specify. If a row of data does not meet the conditions of the `CHECK` constraint, then it is rejected and not inserted into the table.
+
+This can be a very helpful tool for preventing bad data from being inserted into your database, and it can save you a lot of time and hassle later on. In this article, we will show you how to use a `CHECK` constraint in MySQL.
+
+### What is the `CHECK` constraint in MySQL?
+
+MySQL 8.0.16 introduces the `CHECK` constraint, which you can use to protect the data in your tables. A `CHECK` constraint is an integrity constraint used to limit the value range for a column. It ensures that the inserted or updated value in a column must follow certain criteria.
+
+### When do we need `CHECK` constraints?
+
+The following are some examples of when you can use `CHECK` constraints:
+
+- A column must be greater than or equal to a specified value.
+- A date field must be in the format of MM/DD/YYYY.
+- An employee’s joining date must not precede their DOB.
+- The value of an employee’s salary must be a positive integer.
+
+The `CHECK` constraint is commonly used along with the NOT NULL constraint for ensuring that a column contains only valid data and doesn’t contain any NULL values. This is particularly important when defining columns that will be referenced by foreign key constraints.
+
+## Syntax
+
+```sql
+CHECK (column_name value_list)
+```
+
+The `value_list` is a list of values that the `column_name` can have. You can have multiple `value_list`(s) in a single `CHECK` constraint.
+
+If you want to specify multiple `value_list`(s), you need to use the OR keyword between each `value_list`. For example:
+
+```sql
+CHECK (column_name value_list OR value_list)
+```
+
+You can use the AND keyword between each `value_list` to make the `CHECK` constraint more restrictive. For example:
+
+```sql
+CHECK (column_name value_list AND value_list)
+```
+
+The condition for a `value_list` can be any valid MySQL expression. If the condition is not met, the insert or update operation will fail.
+
```

Comment:
I'm not sure what value_list means, but please see the syntax here: https://dev.mysql.com/doc/refman/8.0/en/create-table-check-constraints.html